client/components/oauth/oauth.tsx

```diff
@@ -25,26 +25,34 @@ interface Props {
 }
 
 const OAuth = ({ history }: Props) => {
-  const [cookie, setCookie] = useCookies(['jwt']);
+  const [cookie, setCookie, removeCookie] = useCookies(['jwt']);
 
   const dispatch = useDispatch();
 
   const oAuthSuccess = (response: GoogleLoginResponse) => {
     setCookie('jwt', response.getAuthResponse().id_token, { path: '/' });
-    const decodedJwt = jwtDecode(cookie.jwt);
-    dispatch(setUser(decodedJwt));
-    history.push('/success');
+    if (cookie.jwt) {
+      const decodedJwt = jwtDecode(cookie.jwt);
+      dispatch(setUser(decodedJwt));
+      history.push('/success');
+    }
+    history.push('/');
   };
 
   const oAuthFailure = () => {
     history.push('/');
   };
 
   const checkSignInStatus = () => {
+    const date = new Date();
+    const currentTime = date.getTime() / 1000;
     if (cookie.jwt) {
       const decodedJwt = jwtDecode(cookie.jwt);
-      dispatch(setUser(decodedJwt));
-      return true;
+      if (currentTime < decodedJwt.exp) {
```

Comment:
This function is doing too much. You could start by splitting out `isJwtExpired` and `isSignedInd` and then have a function like `handleAuthStatus` that controls dispatch/render/redirect.